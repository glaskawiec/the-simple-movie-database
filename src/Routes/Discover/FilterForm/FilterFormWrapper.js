import styled from 'styled-components';

const FilterFormWrapper = styled.form`
  margin-top: 1em;
    
  @media (max-width: 576px) {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    flex-wrap: wrap;
  }
`;

export default FilterFormWrapper;
