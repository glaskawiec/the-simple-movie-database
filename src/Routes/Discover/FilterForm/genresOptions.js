const genresOptions = [
  {
    label: 'All',
    value: '',
  },
  {
    label: 'Action',
    value: '28',
  },
  {
    label: 'Animation',
    value: '16',
  },
  {
    label: 'Adventure',
    value: '12',
  },
  {
    label: 'Comedy',
    value: '35',
  },
  {
    label: 'Crime',
    value: '80',
  },
  {
    label: 'Documentary',
    value: '99',
  },
  {
    label: 'Drama',
    value: '18',
  },
  {
    label: 'Horror',
    value: '27',
  },

];

export default genresOptions;
