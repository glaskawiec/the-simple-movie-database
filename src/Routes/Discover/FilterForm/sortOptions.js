const sortOptions = [
  {
    label: 'Popularity Descending',
    value: 'popularity.desc',
  },
  {
    label: 'Popularity Ascending',
    value: 'popularity.asc',
  },
  {
    label: 'Release Date Ascending',
    value: 'release_date.asc',
  },
  {
    label: 'Release Date Descending',
    value: 'release_date.desc',
  },
  {
    label: 'Rating Ascending',
    value: 'vote_average.asc',
  },
  {
    label: 'Rating Descending',
    value: 'vote_average.desc',
  },
];

export default sortOptions;
