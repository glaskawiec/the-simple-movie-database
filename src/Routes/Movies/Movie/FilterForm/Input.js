import styled from 'styled-components';

const Input = styled.input`
color: inherit;
    font: inherit;
    margin: 0;
    line-height: normal;
        -webkit-appearance: initial;
    background-color: initial;
    cursor: default;
    padding: initial;
    border: initial;
        text-rendering: auto;
    letter-spacing: normal;
    word-spacing: normal;
    text-transform: none;
    text-indent: 0px;
    text-shadow: none;
    display: inline-block;
    text-align: start;
}
`;

export default Input;
