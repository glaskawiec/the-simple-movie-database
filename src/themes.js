const Themes = {
  light: {
    colors: {
      primary: '#F2355B',
      secondary: '#161616',
      background: '#FFFFFF',
      text: '#000000',
      mobileMenuOverlay: '#000000',
      card: '#FFFFFF',
      themeToggleThumb: '#161616',
      paginationHover: '#CDCDCD',
    },
  },
  dark: {
    colors: {
      primary: '#F2355B',
      secondary: '#090909',
      background: '#161616',
      text: 'hsla(0,0%,100%,0.88)',
      mobileMenuOverlay: '#FFFFFF',
      card: '#090909',
      themeToggleThumb: '#FFFFFF',
      paginationHover: '#292929',
    },
  },
};

export default Themes;
