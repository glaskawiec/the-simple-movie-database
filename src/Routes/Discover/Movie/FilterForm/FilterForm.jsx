import React from 'react';
import PropTypes from 'prop-types';
import FilterFormWrapper from './FilterFormWrapper';
import Select from './Select/Select';
import TextInput from './TextInput/TextInput';


const yearOptions = [
  {
    label: '2019',
    value: 2019,
  },
  {
    label: '2018',
    value: 2018,
  },
  {
    label: '2017',
    value: 2017,
  },
];

const sortOptions = [
  {
    label: 'Popularity Descending',
    value: 'popularity.desc',
  },
  {
    label: 'Popularity Ascending',
    value: 'popularity.asc',
  },
];

const genresOptions = [
  {
    label: 'All',
    value: 'all',
  },
  {
    label: 'Action',
    value: 'action',
  },
];

const FilterForm = ({
  onYearChange, onSortChange, onGenresChange, onKeywordsChange, year, sort, genres, keywords,
}) => (
  <FilterFormWrapper>
    <TextInput
      onChange={onKeywordsChange}
      placeholder="Filter by keywords..."
      label="Keywords"
      value={keywords}
    />
    <Select
      onChange={onSortChange}
      label="Sort By"
      options={sortOptions}
      value={sort}
    />
    <Select
      onChange={onYearChange}
      label="Year"
      options={yearOptions}
      value={year}
    />

    <Select
      onChange={onGenresChange}
      label="Genres"
      options={genresOptions}
      value={genres}
    />
  </FilterFormWrapper>
);

FilterForm.propTypes = {
  onYearChange: PropTypes.func.isRequired,
  onKeywordsChange: PropTypes.func.isRequired,
  onSortChange: PropTypes.func.isRequired,
  onGenresChange: PropTypes.func.isRequired,
  year: PropTypes.number.isRequired,
  sort: PropTypes.string.isRequired,
  genres: PropTypes.string.isRequired,
  keywords: PropTypes.string.isRequired,
};

export default FilterForm;
