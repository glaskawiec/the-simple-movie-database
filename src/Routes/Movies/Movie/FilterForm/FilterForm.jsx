import React from 'react';
import PropTypes from 'prop-types';
import FilterFormWrapper from './FilterFormWrapper';
import Select from './Select/Select';


const yearOptions = [
  {
    label: '2019',
    value: 2019,
  },
  {
    label: '2018',
    value: 2018,
  },
  {
    label: '2017',
    value: 2017,
  },
];

const sortOptions = [
  {
    label: 'Popularity Descending',
    value: 'popularity.desc',
  },
  {
    label: 'Popularity Ascending',
    value: 'popularity.asc',
  },
];

const genresOptions = [
  {
    label: 'All',
    value: 'all',
  },
  {
    label: 'Action',
    value: 'action',
  },
];

const FilterForm = ({
  onYearChange, onSortChange, onGenresChange, year, sort, genres,
}) => (
  <FilterFormWrapper>
    <Select
      onChange={onYearChange}
      label="Year"
      options={yearOptions}
      value={year}
    />
    <Select
      onChange={onSortChange}
      label="Sort By"
      options={sortOptions}
      value={sort}
    />
    <Select
      onChange={onGenresChange}
      label="Genres"
      options={genresOptions}
      value={genres}
    />
  </FilterFormWrapper>
);

FilterForm.propTypes = {
  onYearChange: PropTypes.func.isRequired,
  onSortChange: PropTypes.func.isRequired,
  onGenresChange: PropTypes.func.isRequired,
  year: PropTypes.number.isRequired,
  sort: PropTypes.string.isRequired,
  genres: PropTypes.string.isRequired,
};

export default FilterForm;
