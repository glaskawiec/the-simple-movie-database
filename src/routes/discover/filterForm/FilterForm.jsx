import React from 'react';
import PropTypes from 'prop-types';
import FilterFormWrapper from './FilterFormWrapper';
import Select from '../../../common/select/Select';
import yearOptions from './yearOptions';
import sortOptions from './sortOptions';
import genresOptions from './genresOptions';

const FilterForm = ({
  onYearChange,
  onSortChange,
  onGenresChange,
  year,
  sort,
  genres,
}) => (
  <FilterFormWrapper>
    <Select
      onChange={onSortChange}
      label="Sort By"
      options={sortOptions}
      value={sort}
    />
    <Select
      onChange={onYearChange}
      label="Year"
      options={yearOptions}
      value={year}
    />

    <Select
      onChange={onGenresChange}
      label="Genres"
      options={genresOptions}
      value={genres}
    />
  </FilterFormWrapper>
);

FilterForm.propTypes = {
  onYearChange: PropTypes.func.isRequired,
  onSortChange: PropTypes.func.isRequired,
  onGenresChange: PropTypes.func.isRequired,
  year: PropTypes.string.isRequired,
  sort: PropTypes.string.isRequired,
  genres: PropTypes.string.isRequired,
};

export default FilterForm;
