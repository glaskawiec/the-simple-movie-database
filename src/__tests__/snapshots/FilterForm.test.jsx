import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import themes from '../../themes';
import FilterForm from '../../routes/discover/filterForm/FilterForm';

test('FilterForm snapshot', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={themes.light}>
        <FilterForm
          genres="snap"
          onGenresChange={jest.fn()}
          onSortChange={jest.fn()}
          onYearChange={jest.fn()}
          sort="snap"
          year="snap"
        />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
