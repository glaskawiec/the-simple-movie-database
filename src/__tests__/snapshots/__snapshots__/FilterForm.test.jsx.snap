// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer snapshot 1`] = `
<form
  className="sc-bdVaJa eTdYLS"
>
  <div
    className="sc-bxivhb iQLHnd"
  >
    <label
      className="sc-htpNat kVGrHg"
    >
      Sort By
    </label>
    <select
      className="sc-bwzfXH SuPFK"
      onChange={[MockFunction]}
      value="snap"
    >
      <option
        value="popularity.desc"
      >
        Popularity Descending
      </option>
      <option
        value="popularity.asc"
      >
        Popularity Ascending
      </option>
      <option
        value="release_date.asc"
      >
        Release Date Ascending
      </option>
      <option
        value="release_date.desc"
      >
        Release Date Descending
      </option>
      <option
        value="vote_average.asc"
      >
        Rating Ascending
      </option>
      <option
        value="vote_average.desc"
      >
        Rating Descending
      </option>
    </select>
  </div>
  <div
    className="sc-bxivhb iQLHnd"
  >
    <label
      className="sc-htpNat kVGrHg"
    >
      Year
    </label>
    <select
      className="sc-bwzfXH SuPFK"
      onChange={[MockFunction]}
      value="snap"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="2019"
      >
        2019
      </option>
      <option
        value="2018"
      >
        2018
      </option>
      <option
        value="2017"
      >
        2017
      </option>
      <option
        value="2016"
      >
        2016
      </option>
      <option
        value="2015"
      >
        2015
      </option>
    </select>
  </div>
  <div
    className="sc-bxivhb iQLHnd"
  >
    <label
      className="sc-htpNat kVGrHg"
    >
      Genres
    </label>
    <select
      className="sc-bwzfXH SuPFK"
      onChange={[MockFunction]}
      value="snap"
    >
      <option
        value=""
      >
        All
      </option>
      <option
        value="28"
      >
        Action
      </option>
      <option
        value="16"
      >
        Animation
      </option>
      <option
        value="12"
      >
        Adventure
      </option>
      <option
        value="35"
      >
        Comedy
      </option>
      <option
        value="80"
      >
        Crime
      </option>
      <option
        value="99"
      >
        Documentary
      </option>
      <option
        value="18"
      >
        Drama
      </option>
      <option
        value="27"
      >
        Horror
      </option>
    </select>
  </div>
</form>
`;
